src/coreComponents/dataRepository/Group.hpp

```diff
@@ -371,132 +296,70 @@ class Group
   ///@{
 
   /**
-   * @brief Retrieve a sub-group from the current Group using an index.
-   * @tparam T type of subgroup
-   * @param[in] index the integral index of the group to retrieve.
-   * @return A pointer to @p T that refers to the sub-group
-   */
-  template< typename T = Group >
-  T * getGroup( localIndex index )
-  {
-    return groupCast< T * >( m_subGroups[index] );
-  }
-
-  /**
-   * @copydoc getGroup(localIndex)
-   */
-  template< typename T = Group >
-  T const * getGroup( localIndex index ) const
-  {
-    return groupCast< T const * >( m_subGroups[index] );
-  }
-
-  /**
-   * @brief Retrieve a sub-group from the current Group using a string.
-   * @tparam T type of subgroup
-   * @param[in] name the name/key of the group lookup and retrieve.
-   * @return A pointer to @p T that refers to the sub-group
+   * @brief Return a pointer to a sub-group of the current Group.
+   * @tparam T The type of subgroup.
+   * @tparam KEY The type of the lookup.
+   * @param key The key used to perform the lookup.
+   * @return A pointer to @p T that refers to the sub-group, if the Group does not exist or it
+   *   has an incompatible type a @c nullptr is returned.
    */
-  template< typename T = Group >
-  T * getGroup( string const & name )
-  {
-    return groupCast< T * >( m_subGroups[name] );
-  }
+  template< typename T = Group, typename KEY = void >
+  T * getGroupPointer( KEY const & key )
+  { return dynamicCast< T * >( m_subGroups[ key ] ); }
 
   /**
-   * @copydoc getGroup(string const &)
+   * @copydoc getGroupPointer(KEY const &)
    */
-  template< typename T = Group >
-  T const * getGroup( string const & name ) const
-  { return groupCast< T const * >( m_subGroups[name] ); }
+  template< typename T = Group, typename KEY = void >
+  T const * getGroupPointer( KEY const & key ) const
+  { return dynamicCast< T const * >( m_subGroups[ key ] ); }
 
   /**
-   * @brief @return Return a reference to the Group @p name.
-   * @tparam The type to return.
-   * @param key The name of the group to retrieve.
-   * @note Will abort if the group doesn't exist.
+   * @brief Return a reference to a sub-group of the current Group.
+   * @tparam T The type of subgroup.
+   * @tparam KEY The type of the lookup.
+   * @param key The key used to perform the lookup.
+   * @return A reference to @p T that refers to the sub-group.
+   * @throw std::domain_error If the Group does not exist is thrown.
    */
-  template< typename T = Group >
-  T & getGroupReference( string const & key )
+  template< typename T = Group, typename KEY = void >
+  T & getGroupReference( KEY const & key )
```

Comment:
I made the change so that `getGroup` would fail with a not found error instead of an "expected a pointer, got a reference" error. I meant to change it back, but I missed it. Thanks.

Reply:
Fixed.